sway/tree/container.c

```diff
@@ -712,6 +712,16 @@ void floating_calculate_constraints(int *min_width, int *max_width,
 
 }
 
+void floating_fix_coordinates(struct sway_container *con, struct wlr_box *old, struct wlr_box *new) {
```

Comment:
If the old box is zeros, e.g. because the transform is unset or taken from fallback output geometry, we should handle it such that the client gets centered like a new window, as it did before.